src/cursor/abstract_cursor.ts

```diff
@@ -711,14 +711,14 @@ function nextDocument<T>(cursor: AbstractCursor): T | null | undefined {
   return null;
 }
 
-function next<T>(cursor: AbstractCursor, blocking: boolean, callback: Callback<T | null>): void {
+function next<T>(cursor: AbstractCursor<T>, blocking: boolean, callback: Callback<T | null>): void {
```

Comment:
question: is adding the generic on AbstractCursor here and below in the call to `nextDocument` required because of TS 4.7 or is it an improvement?

Reply:
An improvement, I could leave it out, the T here serves to type the thing we pass into the callback arg, but a cursor of type T will do just that so connecting them seems right but because of the high-level-ness of this code it doesn't actually care about type T for the cursor.

Reply:
I'm all in favor for this, I just looked at it for a bit to see if it was needed with TS 4.7, that's all!